pkg/auth/providers/oidc/oidc_provider.go

```diff
@@ -555,6 +557,16 @@ func (o *OpenIDCProvider) getGroupsFromClaimInfo(claimInfo ClaimInfo) []v3.Princ
 			groupPrincipals = append(groupPrincipals, groupPrincipal)
 		}
 	}
+
+	// If the claimInfo includes roles, we will treat them like groups
+	// AzureAD uses the roles claim
```

Comment:
A small rephrasing, wdyt?

```suggestion
	// Treat the 'roles' claim as if it were 'groups'.
	// This is done to support identity providers like Azure AD.
```